### ISSUE TITLE
[Question] How do you export TranslateStaticLoader from a SharedModule

### ISSUE THREAD

**Issue by larssn** [ngx-translate/core#388]
So we have a pretty big project (Ionic 2) with lots of smaller ngModules.

We were hoping that you could export a preconfigured TranslateModule like so:

```
export function createTranslateLoader(http: Http) {
    return new TranslateStaticLoader(http, './assets/i18n', '.json');
}

@NgModule({
    declarations: [MomentDatePipe, NoTagsPipe, SpecialCharsPipe],
    exports: [
        IonicModule,
        MomentDatePipe,
        NoTagsPipe,
        SpecialCharsPipe,
        TranslateModule
    ],
    imports: [
        TranslateModule.forRoot({
            provide: TranslateLoader,
            useFactory: (createTranslateLoader),
            deps: [Http]
        })
    ],
    providers: [] // MUST NOT have any providers
})
export class SharedModule { }
```

Noticed that you wrote that one should not use forRoot in shared modules, so I'm aware that this probably is not a good idea, but for the purpose of example...

The translations work when other plugins import the shared module AND it is compiled using JiT.
However, it doesn't seem to work when compiled using AoT - the translations strings dont get translated.

TLDR; can you export a preconfigured TranslateModule from a SharedModule?

Thanks for reading

**Comment by SamVerschueren** [comment]
Are the json translations file being loaded when compiled with AoT?

**Comment by ocombe** [comment]
It works for me but I'm using it in JIT (not AOT). There's a PR for a "forChild" method that would work for lazy loaded + AOT modules, I guess it would also work in your case. But it's a big PR that introduces a lot of changes and I haven't had time to review it yet

**Comment by larssn** [comment]
Thanks guys, will check the network tab first.

Whats the alternative to this PR? Manually configuring every module that needs translations? Just to be clear.

**Comment by SamVerschueren** [comment]
I just wonder. You told us that adding `TranslateModule.forRoot` was just for the sake of the example. I just wonder, why? You should call `TranslateModule.forRoot(config)` in your `AppModule` and use `TranslateModule` in the `SharedModule`.

So this makes me wonder, why the title `How do you export TranslateStaticLoader from a SharedModule`? Shouldn't it be `TranslateStaticLoader not working in AoT`? Because that seems to be the main issue here as the example is incorrect.

So my advice
1. Make sure you call `forRoot` in the main `AppModule`
2. Import `TranslateModule` inside the `SharedModule`
3. Build for AoT
4. Check the network tab that the assets are loaded from the correct path. 

My gut feeling is that it's because the `json` files aren't where they supposed to be. A lot of issues are being opened here because of that. Make sure your build scripts (whatever you use) copy the assets (json translation files) as well and not only the javascript files.

**Comment by larssn** [comment]
I think the docs need clarification then. I was under the assumption that forRoot should only be run in the shared module.

So you're saying that forRoot should only be in the root module? What then, do sub-modules automatically get this pre-configuration if importing TranslationService?

**Comment by SamVerschueren** [comment]
From the [docs](https://github.com/ocombe/ng2-translate#sharedmodule)

> Note: Never call a `forRoot` static method in the `SharedModule`. You will end up with multiple different instances of a service in your injector tree.

Not sure if we could clarify it even more? The examples are also showing this.

> What then, do sub-modules automatically get this pre-configuration if importing TranslationService?

Yes. What happens is that the pre-configuration configures the [providers](https://github.com/ocombe/ng2-translate/blob/master/index.ts#L37-L41). If a sub-module can't find an instance in it's own injector when injecting a service, it will ask the parent if it has an instance for that.

**Comment by ocombe** [comment]
This is fixed in 6.0.0 with the new `forChild` method and the updated documentation :)
